controller/src/main/java/io/pravega/controller/store/stream/records/SealedSegmentsMapShard.java

```diff
@@ -45,18 +45,25 @@
 */
 public class SealedSegmentsMapShard {
     public static final SealedSegmentsMapShardSerializer SERIALIZER = new SealedSegmentsMapShardSerializer();
-    public static final int SHARD_SIZE = 10000;
+    public static final int SHARD_SIZE = 1000;
 
     private final int shardNumber;
     /**
-     * Sealed segments with size at the time of sealing.
-     * segmentId -> sealed segment record.
-     * Each shard contains segments from a range of `segment epoch`.
-     * So each shard size would be say 10k segment numbers. Then the number of records in the map would be 
-     * 10k multiplied by `average number of segments per epoch`.
-     *
-     * So to get sealed segment record -> extract segment epoch from segment id. compute the shard by dividing segment number by 10k.
-     * Fetch the record from the shard.
+     * We maintain a map of Sealed segments with size at the time of sealing. This map can grow very large, O(number of segments)
```

Comment:
A suggested rephrase:

```
     * We maintain a map of Sealed segments with size at the time of sealing. This map can grow very large, 
     * O(number of segments), and hence we will store this in multiple shards. Each shard should ideally
     *  contain a limited number of entries, and we should be able to compute the shard just by looking at
     *  segmentId. A segmentId is composed of two parts, creationEpoch and segmentNumber, which gives
     * us two logical choices for sharding. 
     *
     * If we were to use segmentNumber to identify shard for a segment, then each shard can contain an
     * arbitrarily large number of entries because with rolling transactions, there can be many duplicate
     * segments. If we use creationEpoch, then we can still have an unbounded number of entries because
     * there is no limit on the number of segments created in an epoch. However, it is expected with auto-scale
     * that this number is small for each epoch. 
     *
     * Rolling transactions adds a level of complexity. All segments corresponding to a transaction are recreated
     * in an epoch and number this could be arbitrarily large. We do not foresee transactions in the vast majority
     * of the cases rolling for multiple consecutive epochs, though. As such, assuming a small number of epochs
     * for a rolling transaction and about 1000 segments per epoch on average as an educated guess, we can still 
     * support million segments in each shard with a shard size of 1000. 
     * 
     * Each shard consequently contains segments from a range of epochs in batches of 1000. 
     * To get a sealed segment record -> extract segment epoch from segment id, and compute the shard
     * by taking the modulo of the segment number by the SHARD_SIZE. Fetch the size from the shard map.
```